### PR TITLE
Create LiDAR dataset tutorial

### PR DESCRIPTION
**Pull request by senselessdev1**
## PR Summary
This script generates videos to visualize the AV2 LiDAR Dataset. It may also be used for the Sensor Dataset or the TbV Dataset.

00a6ffc1-6ce9-3bc3-a060-6006e9893a1a_lidar_intensity

## Testing

In order to ensure this PR works as intended, it is:

* [X] unit tested.
* [ ] other or not applicable (*additional detail/rationale required*)

## Compliance with Standards

As the author, I certify that this PR conforms to the following standards:

* [X] Code changes conform to [PEP8](https://www.python.org/dev/peps/pep-0008) and docstrings conform to the Google Python [style guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#38-comments-and-docstrings).
* [X] A well-written summary explains what was done and why it was done.
* [X] The PR is adequately tested and the testing details and links to external results are included.

### CODE REVIEW

File: src/av2/structures/sweep.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
+
+import numpy as np
 
 from av2.geometry.se3 import SE3
 from av2.utils.io import read_ego_SE3_sensor, read_feather
 from av2.utils.typing import NDArrayByte, NDArrayFloat, NDArrayInt
 
 
-@dataclass
 class Sweep:
```
[the remark this answers is not in the document]
Comment (johnwlambert) [reply src/av2/structures/sweep.py:17]: @benjaminrwilson If I recall correctly, the issue was the default values, in the child class. These needed to be provided in the child class, even if the default value was set.

File: src/av2/structures/bev_params.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import math
+from typing import Tuple
+
+import numpy as np
+
+from av2.geometry.sim2 import Sim2
+from av2.utils.typing import NDArrayFloat
+
+
+class BEVParams:
```
Comment (benjaminrwilson) [inline src/av2/structures/bev_params.py:14]: Do you think we could merge functionality with this? https://github.com/argoai/av2-api/blob/082b9b9bc59f29b13d44da6800fd3063adf59f37/src/av2/structures/ndgrid.py#L114

File: src/av2/structures/sweep.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjaminrwilson) [reply src/av2/structures/sweep.py:17]: Huh. I'd really prefer to keep this as a dataset for readability / reduce boiler-plate code if possible. I think we should be able to achieve this functionality using `__post_init__` in the parent class.

Something like:

```
xyz: NDArrayFloat = field(init=False)

...

def __post_init__(self) -> None:
    self.xyz = np.zeros((0, 3), dtype=np.float32)

    ...
```

File: src/av2/utils/io.py [hunk 3]
```diff
@@ -103,8 +103,14 @@ def read_ego_SE3_sensor(log_dir: Path) -> SensorPosesMapping:
 
     Returns:
         Mapping from sensor name to sensor pose.
+
+    Raises:
+        RuntimeError: If no file was found for this log at `calibration/egovehicle_SE3_sensor.feather` on disk.
     """
     ego_SE3_sensor_path = Path(log_dir, "calibration", "egovehicle_SE3_sensor.feather")
+    if not ego_SE3_sensor_path.exists():
+        raise RuntimeError(f"Calibration data could not be found under {log_dir}/calibration/")
```
Comment (jhonykaesemodel) [inline src/av2/utils/io.py:112]: `FileNotFoundError` exception here?

File: tests/structures/test_bev_params.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from av2.utils.typing import NDArrayFloat
+
+
+def test_bev_params() -> None:
+    """Ensure that for a 1000x1000 image, we do not try to index into index=1000."""
+    params = BEVParams(range_m=50, res_meters_per_px=0.1)
+    query: NDArrayFloat = np.array([3.76367188, 49.96875])
+
+    # ymax_m should be 49.9
+    assert params.ymax_m < query[1]
```
Comment (jhonykaesemodel) [inline tests/structures/test_bev_params.py:17]: No newline at EOF.

File: tutorials/generate_bev_lidar_reflectance.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    default=False,
+    help="defaults to false. whether to use a single sweep vs. aggregating points from all sweeps"
+    " using ego-motion compensation",
+    type=bool,
+)
+@click.option(
+    "--interpolate_dense",
+    default=False,
+    help="whether to densely interpolate the reflectance values."
+    "(defaults to false, as very computationally expensive and slow).",
```
Comment (jhonykaesemodel) [inline tutorials/generate_bev_lidar_reflectance.py:235]: I wonder if we really need the text `(defaults to false ...`. It's already explicit in `default=False`.

File: tutorials/generate_bev_lidar_reflectance.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+    "--output_dir",
+    required=True,
+    help="Path to local directory where renderings will be saved.",
+    type=str,
+)
+@click.option(
+    "-l",
+    "--log_id",
+    default="00a6ffc1-6ce9-3bc3-a060-6006e9893a1a",
+    help="unique log identifier.",
```
Comment (jhonykaesemodel) [inline tutorials/generate_bev_lidar_reflectance.py:214]: I'd capitalize the help sentence to make consistent with other options.

File: tutorials/generate_bev_lidar_reflectance.py [hunk 7]
```diff
@@ -0,0 +1,302 @@
+# <Copyright 2022, Argo AI, LLC. Released under the MIT license.>
+
+"""Argoverse 2.0 LiDAR Dataset tutorial.
```
Comment (jhonykaesemodel) [inline tutorials/generate_bev_lidar_reflectance.py:3]: Should we make `Argoverse 2` to be consistent with other files?

File: tutorials/generate_bev_lidar_reflectance.py [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+    bev_refl_img: NDArrayByte = np.zeros((img_h, img_w), dtype=np.uint8)
+
+    if interpolate:
+        bev_refl_img: NDArrayByte = dense_grid_interpolation.interp_dense_grid_from_sparse(
+            grid_img=bev_refl_img,
+            points=img_xy,
+            values=reflectance,
+            grid_h=img_h,
+            grid_w=img_w,
+            interp_method="nearest",  # "linear"
```
Comment (jhonykaesemodel) [inline tutorials/generate_bev_lidar_reflectance.py:79]: I'd remove the comment here.

**Review by jhonykaesemodel — COMMENTED** [review]
Looks great, John! The only thing we'd need to address here is the BEV transform. It seems that we have another BEV class implemented by Ben.

**Comment by benjaminrwilson** [comment]
Future development for this PR is here: https://github.com/argoai/av2-api/pull/32.

**Pull request closed without merging**